partridge/hmm.py

```diff
@@ -0,0 +1,256 @@
+import numpy as np
+from rtree.index import Index
+from shapely.geometry import LineString, Point
+
+from partridge.geo import (
+    bbox_from_point,
+    great_circle_vec,
+    pairwise_great_circle_vec,
+)
+from partridge.utilities import (
+    cached_property,
+    ddict,
+    ddict2dict,
+    dict_argmax,
+    lru_cache,
+    pairwise,
+)
+
+
+SQRT2PI = np.sqrt(2.0 * np.pi)
+
+# Meticulously hand-tuned parameters!
+SIGMA = 2.25
+BETA = 18.797498
+
+
+class NoPathError(RuntimeError):
+    pass
+
+
+def shape_dist_traveled(stop_lons, stop_lats, shape_lons, shape_lats,
+                        threshold=200, sigma=SIGMA, beta=BETA):
```

Comment:
I am assuming this is a 200 meter threshold, correct? 

If so, then the process appears to be: 
Take a point, buffer by 200m, and get bounds. Check if bounds intersect with the spatial index developed, which is based on the indexed bounds of each line segment from the route.

So a 2 part question (1 related to how the index is being created, which I will add down below) but, here: 

Should not the buffer length be variable, and a function of the nature of the LineStrings generated? That is, dense networks should have greater sensitivity (e.g. smaller threshold values) whereas comparatively sparse networks can accomodate (perhaps even need) larger threshold values.

Acknowledging this could be a stretch goal for mvp feature and first pass may find threshold default 200 to be acceptable.